sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.Emitter.cs

```diff
@@ -0,0 +1,104 @@
+
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+internal partial class FunctionExecutorGenerator
+{
+    internal static class Emitter
+    {
+        internal static string Emit(IEnumerable<ExecutableFunction> functions, CancellationToken cancellationToken)
+        {
+            string result = $$"""
+                         // <auto-generated/>
+                         using System;
+                         using System.Threading.Tasks;
+                         using Microsoft.Extensions.Hosting;
+                         using Microsoft.Extensions.DependencyInjection;
+                         using Microsoft.Azure.Functions.Worker.Context.Features;
+                         using Microsoft.Azure.Functions.Worker.Invocation;
+                         namespace Microsoft.Azure.Functions.Worker
+                         {
+                             internal class DirectFunctionExecutor : IFunctionExecutor
+                             {
+                                 public async ValueTask ExecuteAsync(FunctionContext context)
+                                 {
+                                     {{GetMethodBody(functions)}}
+                                 }
+                             }
+                             public static class FunctionExecutorHostBuilderExtensions
+                             {
+                                 ///<summary>
+                                 /// Configures an optimized function executor to the invocation pipeline.
+                                 ///</summary>
+                                 public static IHostBuilder ConfigureGeneratedFunctionExecutor(this IHostBuilder builder)
+                                 {
+                                     return builder.ConfigureServices(s => 
+                                     {
+                                         s.AddSingleton<IFunctionExecutor, DirectFunctionExecutor>();
+                                     });
+                                 }
+                             }
+                         }
+                         """;
+
+            return result;
+        }
+
+        private static string GetMethodBody(IEnumerable<ExecutableFunction> functions)
+        {
+            var sb = new StringBuilder();
+            sb.Append(@"var inputBindingFeature = context.Features.Get<IFunctionInputBindingFeature>()!;
+            var inputBindingResult = await inputBindingFeature.BindFunctionInputAsync(context)!;
+            var inputArguments = inputBindingResult.Values;
+            ");
+            foreach (ExecutableFunction function in functions)
+            {
+                sb.Append($@"
+            if (string.Equals(context.FunctionDefinition.EntryPoint, ""{function.EntryPoint}"", StringComparison.OrdinalIgnoreCase))
+            {{");
+
+                int constructorParamCounter = 0;
```

Comment:
The activation process should be consistent with what we have today. That will also handle constructor selection and type injection (i.e., continue to go through `IFunctionActivator`).

Expansion to optimize activation can use the existing extensibility, but should be done separately

Reply:
Good point. Updated to use IFunctionActovator.